build-yarn/action.yml

```diff
@@ -0,0 +1,139 @@
+---
+name: Build Yarn
+description: GitHub Action to build, analyze, and deploy a Yarn project with SonarQube integration
+inputs:
+  public:
+    description: Whether to build and deploy with/to public repositories. Set to `true` for public repositories (OSS), `false` for private.
+    default: ${{ github.event.repository.visibility == 'public' && 'true' || 'false' }}
+  artifactory-reader-role:
+    description: Suffix for the Artifactory reader role in Vault. Defaults to `private-reader` for private repositories, and `public-reader`
+      for public repositories.
+  artifactory-deployer-role:
+    description: Suffix for the Artifactory deployer role in Vault. Defaults to `qa-deployer` for private repositories, and
+      `public-deployer` for public repositories.
+  artifactory-deploy-repo:
+    description: Name of deployment repository
+    required: true
+  artifactory-deploy-access-token:
+    description: Access token to deploy to Artifactory
+    required: true
```

Comment:
Not a parameter. This comes from the role, and the action will request the vault:
```
development/artifactory/token/{REPO_OWNER_NAME_DASH}-${{ env.ARTIFACTORY_DEPLOYER_ROLE }} access_token | ARTIFACTORY_DEPLOY_ACCESS_TOKEN
```